docs/design/network-policy-support.md

```diff
@@ -0,0 +1,247 @@
+
+## Multicluster Network Policy support
+
+This document describes the Network Policy use cases that we propose
+supporting for a first iteration in Submariner. In this iteration, the
+intention is to support the ingress side of existing k8s Network Policies
+as is, while in the future we could explore with the kubernetes community
+options to extend Network Policies for multi cluster scenarios.
+
+> A Kubernetes, network policy is a specification of how pods are allowed
+> to communicate with each other and other network endpoints. It is like
+> Access Control Lists using which we can specify the traffic that would
+> be allowed/denied to the PODs. Network Policies are additive, so having
+> two Network Policies that select the same Pods will result in allowing
+> both defined policies.
+
+
+> Network Policies are implemented by the networking plugin which is the
+> CNI for the K8s cluster. If the underlying network plugin does not
```

Comment:
How about "by the underlying  networking CNI plugin for the k8s cluster"?

Also the first statement states Network Policies are implemented by the CNI but the second statement sort of contradicts that by indicating it may not. I think it would be clearer to say "Network Policies are assumed to be implemented by the...".

Reply:
Yeah, we are trying to explain that some CNIs don't implement it. I'll try to rephrase.